### PR TITLE
feat: implement file upload

### PR DESCRIPTION
**Pull request by n-r-b**
### JIRA link
https://companieshouse.atlassian.net/browse/LFA-1278

### Change description
Sub story 1

### Work checklist

- [ ] Tests added where applicable
- [ ] UI changes look good on mobile
- [ ] UI changes meet accessibility criteria

### Merge instructions

We are commited to keeping commit history clean, consistent and linear. To achive this commit should be structured as follows:

```
<type>[optional scope]: <description>
```

and contain the following structural elements:

- fix: a commit that patches a bug in your codebase (this correlates with PATCH in semantic versioning),
- feat: a commit that introduces a new feature to the codebase (this correlates with MINOR in semantic versioning),
- BREAKING CHANGE: a commit that has a footer `BREAKING CHANGE:` introduces a breaking API change (correlating with MAJOR in semantic versioning). A BREAKING CHANGE can be part of commits of any type,
- types other than `fix:` and `feat:` are allowed, for example `build:`, `chore:`, `ci:`, `docs:`, `style:`, `refactor:`, `perf:`, `test:`, and others,
- footers other than `BREAKING CHANGE: <description>` may be provided.

### CODE REVIEW

File: src/controllers/index.ts [hunk 1]
```diff
@@ -1,6 +1,7 @@
 import 'app/controllers/CheckYourAppealController';
 import 'app/controllers/ConfirmationController';
 import 'app/controllers/EntryController'
+import 'app/controllers/EvidenceUploadController';
```
Comment (geek-soft) [inline src/controllers/index.ts:4]: Even smarter might be not register controllers when feature is off compared to use of middleware :) To be tried in the future.

File: src/middleware/FileTransferFeatureMiddleware.ts [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+import { loggerInstance } from 'app/middleware/Logger';
+import { getEnv } from 'app/utils/EnvironmentUtils';
+import { ENTRY_PAGE_URI } from 'app/utils/Paths';
+
+@provide(FileTransferFeatureMiddleware)
+export class FileTransferFeatureMiddleware extends BaseMiddleware {
+
+    public handler: RequestHandler = (req: Request, res: Response, next: NextFunction): void => {
+
+        loggerInstance().info(`${FileTransferFeatureMiddleware.name} - handler: request url is ` + req.url);
```
Comment (geek-soft) [inline src/middleware/FileTransferFeatureMiddleware.ts:14]: @n-r-b either you use string interpolation e.g. `${xxx} - ` or string concatenation e.g. `xxx + ' - '` - no point using both.

Comment (geek-soft) [reply src/middleware/FileTransferFeatureMiddleware.ts:14]: Another thing, this log is useless.

File: src/controllers/index.ts (hunk 1 above)
Comment (n-r-b) [reply src/controllers/index.ts:4]: great point! 👍

File: src/middleware/FileTransferFeatureMiddleware.ts (hunk 2 above)
Comment (geek-soft) [reply src/middleware/FileTransferFeatureMiddleware.ts]: The reason I am saying useless is that nobody cares whether URI covered by middleware is hit. It is much useful to know that URI who is disabled was triggered.

We need to look at our logging at some point to make it more useful.

**Pull request merged**